cmd/spire-server/cli/run/run.go

```diff
@@ -61,22 +61,24 @@ type Config struct {
 }
 
 type serverConfig struct {
-	BindAddress         string             `hcl:"bind_address"`
-	BindPort            int                `hcl:"bind_port"`
-	CAKeyType           string             `hcl:"ca_key_type"`
-	CASubject           *caSubjectConfig   `hcl:"ca_subject"`
-	CATTL               string             `hcl:"ca_ttl"`
-	DataDir             string             `hcl:"data_dir"`
-	Experimental        experimentalConfig `hcl:"experimental"`
-	JWTIssuer           string             `hcl:"jwt_issuer"`
-	LogFile             string             `hcl:"log_file"`
-	LogLevel            string             `hcl:"log_level"`
-	LogFormat           string             `hcl:"log_format"`
-	RegistrationUDSPath string             `hcl:"registration_uds_path"`
-	DeprecatedSVIDTTL   string             `hcl:"svid_ttl"`
-	DefaultSVIDTTL      string             `hcl:"default_svid_ttl"`
-	TrustDomain         string             `hcl:"trust_domain"`
-	UpstreamBundle      *bool              `hcl:"upstream_bundle"`
+	BindAddress         string                        `hcl:"bind_address"`
+	BindPort            int                           `hcl:"bind_port"`
+	BundleEndpoint      bundleEndpointConfig          `hcl:"bundle_endpoint"`
+	CAKeyType           string                        `hcl:"ca_key_type"`
+	CASubject           *caSubjectConfig              `hcl:"ca_subject"`
+	CATTL               string                        `hcl:"ca_ttl"`
+	DataDir             string                        `hcl:"data_dir"`
+	Experimental        experimentalConfig            `hcl:"experimental"`
+	FederateWith        map[string]federateWithConfig `hcl:"federate_with"`
```

Comment:
I forgot about https://github.com/hashicorp/hcl/issues/322#issuecomment-561919807. One consequence of moving map into the main section is that we can no longer rely on the `unusedKeys` field for detecting misconfiguration since it does not work correctly with the sections containing maps.

I wonder if we should move all of the federation stuff under a `federation` section:
```
federation {
     bundle_endpoint {
     }
     federate_with "A" {
     }
     federate_with "B" {
    }
}

Alternatively, we could just forego using a map here and just do a list where each item also includes a value with the trust domain. We could check for duplicates during validation.

Reply:
> I wonder if we should move all of the federation stuff under a federation section:

I was wondering the same yesterday. I think it would be nice to group the federation related configuration inside its own section.
> Alternatively, we could just forego using a map here and just do a list where each item also includes a value with the trust domain. We could check for duplicates during validation.

Wasn't aware of the hcl issue you mentioned. So, this suggestion makes sense to me.